build_resolvers/lib/src/build_asset_uri_resolver.dart

```diff
@@ -240,9 +257,17 @@ Set<AssetId> _parseDirectives(String content, AssetId from) => HashSet.of(
           .map((content) => AssetId.resolve(Uri.parse(content), from: from)),
     );
 
+/// A resource which gives access to the cached dependencies of assets based on
+/// parsing the directives.
+final dependenciesResource = Resource<
+        Future<Iterable<AssetId>?> Function(AssetId id, BuildStep buildStep)>(
+    () => (AssetId id, BuildStep buildStep) => BuildAssetUriResolver.instance
```

Comment:
Since this touches global state and doesn't use `dispose` or `beforeExit` I don't think we are gaining anything from `Resource`.

Would it make sense to replace the usage with a direct definition?

```dart
Future<Iterable<AssetId>?> dependenciesOf(
        AssetId id, BuildStep buildStep) async =>
    (await BuildAssetUriResolver.instance
            ._updateCachedAssetState(id, buildStep))
        ?.dependencies;
```

Reply:
Good point, done!